addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3814,20 +3814,23 @@ registry.BackgroundImage = SnippetOptionWidget.extend({
     /**
      * @override
      */
-    _computeWidgetState: function (methodName) {
+    _computeWidgetState: function (methodName, params) {
         switch (methodName) {
             case 'background':
                 return getBgImageURL(this.$target[0]);
             case 'dynamicColor':
-                return new URL(getBgImageURL(this.$target[0]), window.location.origin).searchParams.get('c1');
+                return new URL(getBgImageURL(this.$target[0]), window.location.origin).searchParams.get(params.colorName);
         }
         return this._super(...arguments);
     },
     /**
      * @override
      */
     _computeWidgetVisibility(widgetName, params) {
-        if (widgetName === 'dynamic_color_opt') {
+        if ('colorName' in params) {
+            const src = getBgImageURL(this.$target[0]);
+            return src.match('[\\?|\\&]' + params.colorName + '=') !== null;
```

Comment:
same here as for the other one: `new URL(...).searchParams.has(...)` conveys intent better than a regex match.